File: javascript/ts/org/noear/socketd/transport/core/ChannelAssistant.ts
import {Frame} from "./Frame";

/**
 * 通道助理
 *
 * @author noear
 * @since 2.0
 * @class
 */
export interface ChannelAssistant<T> {
    /**
     * 写入
     *
     * @param {*} target 目标
     * @param {org.noear.socketd.transport.core.Frame} frame  帧
     */
    write(target: T, frame: Frame);

    /**
     * 是否有效
     * @param {*} target
     * @return {boolean}
     */
    isValid(target: T): boolean;

    /**
     * 关闭
     * @param {*} target
     */
    close(target: T);

    /**
     * 获取远程地址
     * @param {*} target
     * @return {java.net.InetSocketAddress}
     */
    getRemoteAddress(target: T): string;

    /**
     * 获取本地地址
     * @param {*} target
     * @return {java.net.InetSocketAddress}
     */
    getLocalAddress(target: T): string;
}

File: java/socketd/src/main/java/org/noear/socketd/transport/core/ChannelAssistant.java
package org.noear.socketd.transport.core;

import org.noear.socketd.utils.IoCompletionHandler;

import java.io.IOException;
import java.net.InetSocketAddress;

/**
 * 通道助理
 *
 * @author noear
 * @since 2.0
 */
public interface ChannelAssistant<T> {
    /**
     * 写入
     *
     * @param target  目标
     * @param frame   帧
     * @param channel 通道
     */
    void write(T target, Frame frame, ChannelInternal channel, IoCompletionHandler completionHandler);

    /**
     * 是否有效
     */
    boolean isValid(T target);

    /**
     * 关闭
     */
    void close(T target) throws IOException;

    /**
     * 获取远程地址
     */
    InetSocketAddress getRemoteAddress(T target) throws IOException;

    /**
     * 获取本地地址
     */
    InetSocketAddress getLocalAddress(T target) throws IOException;
}